style/swift.md

```diff
@@ -1,3 +1,1037 @@
+
+Classes have [reference semantics](https://developer.apple.com/library/mac/documentation/Swift/Conceptual/Swift_Programming_Language/ClassesAndStructures.html#//apple_ref/doc/uid/TP40014097-CH13-XID_145). Use classes for things that do have an identity or a specific life cycle. You would model a ViewModel as a class because two ViewModel objects might be in a different state. Although they might represent the same screen, it doesn't mean they have the same state (e.g. `.loading` vs `.loaded([T])`.
+
+### Use of Self
+
+For conciseness, avoid using `self` since Swift does not require it to access an object's properties or invoke its methods.
+
+Use self only when required by the compiler (in `@escaping` closures, or in initializers to disambiguate properties from arguments). In other words, if it compiles without `self` then omit it.
+
+
+### Computed Properties
+
+For conciseness, if a computed property is read-only, omit the get clause. The get clause is required only when a set clause is provided. For single-expressions where the context is clear, both explicit and implicit returns can be used:
+
+**Preferred**:
+```swift
+var diameter: Double { radius * 2 }
+
+// Or
+
+var diameter: Double {
+  return radius * 2
+}
+```
+
+**Not Preferred**:
+```swift
+var diameter: Double {
+  get {
+    return radius * 2
+  }
+}
+```
+
+### Final
+
+Mark classes or members as `final` when appropriate. In the below example, `Box` has a particular purpose and customization in a derived class is not intended. Marking it `final` makes that clear.
+
+```swift
+// Turn any generic type into a reference type using this Box class.
+final class Box<T> {
+  let value: T
+  init(_ value: T) {
+    self.value = value
+  }
+}
+```
+
+## Functions
+
+### Declarations
+
+Keep short function declarations on one line including the opening brace:
+
+```swift
+func reticulateSplines(spline: [Double]) -> Bool {
+  // reticulate code goes here
+}
+```
+
+For functions with long signatures, more than the recommended length, put each parameter on a new line and add an extra indent on subsequent lines.
+
+```swift
+func reticulateSplines(
+    spline: [Double], 
+    adjustmentFactor: Double,
+    translateConstant: Int, 
+    comment: String
+) -> Bool {
+  // reticulate code goes here
+}
+```
+
+Don't use `(Void)` to represent the lack of an input; simply use `()`.
+Use `Void` instead of `()` for closure outputs.
+Don't use `Void` for function outputs as it is redundant.
+
+**Preferred**:
+
+```swift
+func updateConstraints() {
+  // magic happens here
+}
+
+typealias CompletionHandler = (result) -> Void
+```
+
+**Not Preferred**:
+
+```swift
+func updateConstraints() -> () {
+  // magic happens here
+}
+
+typealias CompletionHandler = (result) -> ()
+```
+
+For single-expression functions where the context is clear, both explicit and implicit returns can be used:
```

Comment:
It's tempting to try and aim for consistency by being more strict here. e.g. `where implicit returns are possible they should be used unless apparent it will cause confusion.`